### PR TITLE
[ArrayPractice] Investigate arrays allocation

### PR DESCRIPTION
**Pull request by plotnikov01**

### CODE REVIEW

File: src/array_practice.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#include <stdio.h>
+
+int main()
```
Comment (MLGMAG) [inline src/array_practice.c:3]: Change it to `int main() {`

File: src/array_practice.c [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+{
+	int numbers[] = {1, 4, 5, 6, 10};
+	char *strings[] = {"Hello", "World", "or", "Hello", "Sasha"};
+	int x;
+	int sum = 0;
+
+	printf("Enter number x for multiply an array : ");
+	scanf("%d", &x);
+
+	for(int i = 0; i < sizeof(numbers) / sizeof(numbers[0]); i++) {
```
Comment (MLGMAG) [inline src/array_practice.c:13]: Follow the requirements:
1. Create a function, that will receive an array of 5 integers, sum them and return a result.
2. Create a function, that will receive an array of 5 integers, multiply them by x (input parameter) and return them.

**Review by MLGMAG — CHANGES_REQUESTED** [review]

**Comment by MLGMAG** [comment]
Change the commit message

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: src/array_practice.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+        return res;
+}
+
+int main() {
+	int numbers[] = {1, 4, 5, 6, 10};
+	char *strings[] = {"Hello", "World", "or", "Hello", "Sasha"};
+
+	int numbersCount = sizeof(numbers) / sizeof(numbers[0]);
+    	int stringsCount = sizeof(strings) / sizeof(strings[0]);
```
Comment (MLGMAG) [inline src/array_practice.c:32]: Extract the size calculation into separate function

File: src/array_practice.c [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+	int numbersSum = sumArray(numbers);
+	int numbersMultiply =  multiplyArray(numbers);
```
Comment (MLGMAG) [inline src/array_practice.c:35]: Format code

File: src/array_practice.c [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+	printf("Sum of numbers array: %d \n", numbersSum);
+	printf("Result of multiply numbers array:  %d \n", numbersMultiply);
+
+	for(int i = 0; i < numbersCount; i++) {
```
Comment (MLGMAG) [inline src/array_practice.c:40]: Extract the array print into separate function

File: src/array_practice.c [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	int sum = 0;
+
+	for(int i = 0; i < 5; i++) {
+		sum += arr[i];
+	}
+
+	return sum;
+}
+
+int multiplyArray(int arr[5]) {
```
Comment (MLGMAG) [inline src/array_practice.c:13]: What happens, if array has more then or less then 5 elements?

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: src/array_practice.c (hunk 3 above)
Comment (plotnikov01) [reply src/array_practice.c:32]: Can I use macros for that?

#define LENGTH(a) sizeof(a) / sizeof(a[0])

Comment (MLGMAG) [reply src/array_practice.c:32]: I have implemented it in this way and got 0 errors / warnings.

image

File: src/array_practice.c (hunk 6 above)
Comment (MLGMAG) [reply src/array_practice.c:13]: When it is better to specify the exact amount?
Why this feature exists?

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: src/array_practice.c (hunk 3 above)
Comment (plotnikov01) [reply src/array_practice.c:32]: It works for int array, should I use separate functions for chars and ints?

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: src/array_practice.c [hunk 7]
```diff
@@ -0,0 +1,58 @@
+#include <stdio.h>
+
+#define SIZEOF(arr) sizeof(arr) / sizeof(*arr)
+
+int sumArray(int arr[], int arrCount) {
```
[the remark this answers is not in the document]
Comment (plotnikov01) [reply src/array_practice.c]: done

**Review by MLGMAG — APPROVED** [review]

**Pull request merged**